app/controllers/system_notifications_controller.rb

```diff
@@ -29,21 +30,19 @@ def mark
 
   def prepare_notifications
     page = (params[:page] || 1).to_i
-    system_notifications = SystemNotification.last_notifications(current_user, params[:q])
+    system_notifications = SystemNotification.last_notifications(current_user, params[:search_queue])
                                              .page(page).per(Constants::ACTIVITY_AND_NOTIF_SEARCH_LIMIT)
     unless system_notifications.blank? || system_notifications.last_page?
       more_url = url_for(
         system_notifications_url(
           format: :json,
-           page: page + 1,
-           last_notification: system_notifications.last.id
+          page: page + 1
         )
       )
     end
-    {
+    @system_notifications = {
       notifications: system_notifications,
-      more_notifications_url: more_url,
-      page: page
+      more_notifications_url: more_url || nil
```

Comment:
Are you sure this will work? If `more_url` isn't defined, this will still fail, if I'm not mistaken. Hence, it should be defined before the `unless` statement, i.e. `more_url = nil`.

Reply:
He will create this variable anyway with nil value, here is example - https://stackoverflow.com/questions/27592601/ruby-variables-defined-in-if-else-statement-are-accessible-outside-of-if-else